api.go

```diff
@@ -397,15 +397,21 @@ func (c Client) executeMethod(method string, metadata requestMetadata) (res *htt
 		res, err = c.do(req)
 		if err != nil {
 			// Fail quickly here, no need to retry.
-			return nil, err
+			return res, err
 		}
```

Comment:
nil is returned deliberately when error is set. It is the correct coding style. We cannot assume people write clean code with proper error checks all the time.